client/components/LayoutEditor/LayoutEditorPubs.tsx

```diff
@@ -0,0 +1,331 @@
+	);
+
+	const setHideDates = useCallback(
+		(nextHideDates: boolean) =>
+			onChange({
+				hideDates: nextHideDates,
+			}),
+		[onChange],
+	);
+
+	const setHideContributors = useCallback(
+		(nextHideContributors: boolean) =>
+			onChange({
+				hideContributors: nextHideContributors,
+			}),
+		[onChange],
+	);
+
+	const setHideEdges = useCallback(
+		(nextHideEdges: boolean) =>
+			onChange({
+				hideEdges: nextHideEdges,
+			}),
+		[onChange],
+	);
+
+	const setCollectionIds = useCallback(
+		(nextCollectionIds: string[]) => {
+			const validPubIds = (pubIds || [])
+				.map((pubId) => pubsById[pubId])
+				.filter((pub) => pub && pubIsInCollections(pub, nextCollectionIds))
+				.map((pub) => pub.id);
+			onChange({
+				collectionIds: nextCollectionIds,
+				pubIds: validPubIds,
+			});
+		},
+		[onChange, pubIds, pubsById],
+	);
+
+	const setPubIds = useCallback(
+		(pubs: Pub[]) => onChange({ pubIds: pubs.map((pub) => pub.id) }),
+		[onChange],
+	);
+
+	const setTitle = useCallback((title: string) => onChange({ title: title }), [onChange]);
+
+	const renderTitle = () => {
+		return (
+			<InputField
+				label="Title"
+				value={block.content.title}
+				onChange={(evt) => setTitle(evt.target.value)}
+			/>
+		);
+	};
+
+	const renderCollectionFilter = () => {
+		const { collections } = communityData;
+		return (
+			<InputField label="Filter by Collection">
+				<CollectionMultiSelect
+					allCollections={collections}
+					selectedCollectionIds={collectionIds}
+					onItemSelect={(newCollectionId) =>
+						setCollectionIds([...collectionIds, newCollectionId])
+					}
+					onRemove={(_, collectionIndex) => {
+						const nextCollectionIds = [...collectionIds];
+						nextCollectionIds.splice(collectionIndex, 1);
+						setCollectionIds(nextCollectionIds);
+					}}
+					placeholder="Add collections..."
+				/>
+			</InputField>
+		);
+	};
+
+	const renderLimit = () => {
+		return (
+			<InputField label="Limit">
+				<div className="bp3-button-group bp3-select">
+					<select
+						value={limit}
+						onChange={(evt) => setLimit(parseInt(evt.target.value, 10))}
+					>
+						<option value={0}>Show All pubs</option>
+						{limitPubsOptions.map((item) => {
+							return (
+								<option value={item} key={`option-${item}`}>
+									Show {item} pub{item === 1 ? '' : 's'}
+								</option>
+							);
+						})}
+					</select>
+				</div>
+			</InputField>
+		);
+	};
+
+	const renderOrder = () => {
+		return (
+			<InputField label="Order">
+				<div className="bp3-button-group bp3-select">
+					<select
+						value={sort}
+						onChange={(evt) => setSort(evt.target.value as PubSortOrder)}
+					>
+						<option value="creation-date">Creation date</option>
+						<option value="collection-rank">Collection order</option>
+					</select>
+				</div>
+			</InputField>
+		);
+	};
+
+	const renderPinnedPubs = () => {
+		return (
+			<InputField label="Pinned Pubs">
+				<Popover
+					content={
+						<OrderPicker<Pub>
```

Comment:
Gotta be honest, I've never seen this syntax before! Was TypeScript unable to infer the type argument from the props, e.g. `allItems`?

Reply:
Actually it looks like it can infer it!